### ISSUE TITLE
Auto resize parent in nested grid

### ISSUE THREAD

**Issue by ptorrent** [gridstack/gridstack.js#2098]
Hello there !

Is there a way to resize parent div instead of adding scroll when I resize a child item ?

![flash2](https://user-images.githubusercontent.com/6083644/201483821-378ec9e1-7118-4eda-a9f1-cf9a333c13db.gif)

Something like that :

![flash2](https://user-images.githubusercontent.com/6083644/201483873-b842bc35-3723-4845-8ffd-d03e948c408b.gif)

I searched in the documentation and examples but I did not find anything about that.

Sorry for posting question there but I cannot join slack ?

https://join.slack.com/t/gridstackjs/shared_invite/zt-1hb1k3mhh-WR6YyYjkUbMw7J6RqDwPRQ

(link on the main page..)

After Reading other threads, looks like a common issue.. I can do donation for help about that !

Thanks !

**Comment by adumesny** [comment]
this would be a new feature - not trivial because while you grow the parent, it itself has to push other items out of the way (unlike your example 2) and if it can grow, it should also shrink back, but then you need to leave the original space you have in example 1 and not just shrink to fit...

Can be done with right donation... I'll fix slack (for some reason I can only do 30 day invites as I don't own that channel yet)

**Comment by ptorrent** [comment]
Hello, thanks for your answer and the slack link. How can I pm you?

**Comment by ptorrent** [comment]
I don't know if it's a good approach but I was able to do it by calling this function any time a grid widget is resizing, added or removed.

```
 function resizeGridHeight (SubGrid){
	var rowCount = SubGrid.getRow() + 1
			
	this.Grid.update(SubGrid._isNested.el,{
		h    : rowCount,
		minH : rowCount
	})
}
```

```
SubGrid.on('resize', function(e,el){
	this.resizeGridHeight($subGrid)
}.bind(this))
```

By the way I add this in css:
```
.grid-stack-item-content{
	overflow-y:hidden !important;
}
```


I think i can change "+1" by calculating the margin

**Comment by adumesny** [comment]
glad you found a workaround for your case, but you are using internal vars so no guarantee it won't break in the future (and you are using JS which won't catch it). The change wouldn't work in the lib - you don't remember initial extra space (resize parent, then resize child and you loose those changes), nor recursive, nor handle locks items that may need below, etc...

**Comment by adumesny** [comment]
fixed in 9.2
